x/erc20/keeper/precompile.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/evmos/evmos/v15/precompiles/erc20"
+	"github.com/evmos/evmos/v15/precompiles/werc20"
+	"github.com/evmos/evmos/v15/x/erc20/types"
+)
+
+// RegisterERC20Extensions registers the ERC20 precompiles with the EVM.
+func (k Keeper) RegisterERC20Extensions(ctx sdk.Context) error {
+	precompiles := make([]vm.PrecompiledContract, 0)
+	params := k.evmKeeper.GetParams(ctx)
+	evmDenom := params.EvmDenom
+	logger := ctx.Logger()
+
+	k.IterateTokenPairs(ctx, func(tokenPair types.TokenPair) bool {
+		// skip registration if token is native or if it has already been registered
+		// NOTE: this should handle failure during the selfdestruct
+		if tokenPair.ContractOwner != types.OWNER_MODULE ||
+			params.IsPrecompileRegistered(tokenPair.Erc20Address) {
+			return false
+		}
```

Comment:
The `IsPrecompileRegistered` checks if the precompile is in the _active_ precompiles, but it could have been already registered and then de-activated? Is that a valid case for the ERC20 precompiles?

Reply:
good point. Let's rename it to `IsActivePrecompile(address string)` and create another function that checks if the precompile is available on the EVM keeper (`k.IsAvailablePrecompile(address string)`)

Reply:
Addressed 👍